### PR TITLE
[RLlib] Bring categorical distribution temperature and huber loss fn to CRR algorithm.

### PR DESCRIPTION
**Pull request by gjoliver**
## Why are these changes needed?

Feature parity between DQN and CRR.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [*] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/algorithms/crr/torch/crr_torch_policy.py [hunk 1]
```diff
@@ -389,14 +407,15 @@ def _compute_critic_loss(
         ).squeeze(-1)
 
         # compute the MSE loss for all q-functions
-        loss_q1 = (target - q1) ** 2
-        loss_q2 = (target - q2) ** 2
-        loss = 0.5 * (loss_q1 + loss_q2)
+        loss_q1 = target - q1
+        loss_q2 = target - q2
+        loss_fn = l2_loss if self.config["td_error_loss_fn"] == "mse" else huber_loss
+        loss = loss_fn(torch.hstack((loss_q1, loss_q2)))
```
Comment (kouroshHakha) [inline rllib/algorithms/crr/torch/crr_torch_policy.py:413]: nit: can you use .stack with (dim=0 or 1).

File: rllib/algorithms/crr/torch/crr_torch_policy.py [hunk 2]
```diff
@@ -389,14 +407,15 @@ def _compute_critic_loss(
         ).squeeze(-1)
 
         # compute the MSE loss for all q-functions
-        loss_q1 = (target - q1) ** 2
-        loss_q2 = (target - q2) ** 2
-        loss = 0.5 * (loss_q1 + loss_q2)
+        loss_q1 = target - q1
```
Comment (kouroshHakha) [inline rllib/algorithms/crr/torch/crr_torch_policy.py:410]: since we are here, can we just unify the naming with DQN? loss_q1 --> td_error_q1, etc.
The sign will tell us about under or overfitting. and it should be q1 - target, similar to DQN's definition.

File: rllib/algorithms/crr/torch/crr_torch_policy.py [hunk 3]
```diff
@@ -389,14 +407,15 @@ def _compute_critic_loss(
...
+        loss_q2 = target - q2
+        loss_fn = l2_loss if self.config["td_error_loss_fn"] == "mse" else huber_loss
+        loss = loss_fn(torch.hstack((loss_q1, loss_q2)))
         loss = loss.mean(0)
 
         # logging
-        self.log("loss_q1", loss_q1.mean())
-        self.log("loss_q2", loss_q2.mean())
+        self.log("loss_q1", (loss_q1 ** 2).mean())
```
Comment (kouroshHakha) [inline rllib/algorithms/crr/torch/crr_torch_policy.py:417]: nit: also plot td_errors

File: rllib/models/tf/tf_action_dist.py [hunk 4]
```diff
@@ -97,6 +97,16 @@ def required_model_output_shape(action_space, model_config):
         return action_space.n
 
 
+def get_categorical_with_temperature(t: float):
```
Comment (kouroshHakha) [inline rllib/models/tf/tf_action_dist.py:100]: nit: Can we update the name to indiciate this is returning a class? get_categorical_class_with_temperature?

**Review by kouroshHakha — APPROVED** [review]
Looks good. Only a few subjective nits. Feel free to ignore them.

File: rllib/models/tf/tf_action_dist.py (hunk 4 above)
Comment (gjoliver) [reply rllib/models/tf/tf_action_dist.py:100]: done

File: rllib/algorithms/crr/torch/crr_torch_policy.py (hunk 1 above)
Comment (gjoliver) [reply rllib/algorithms/crr/torch/crr_torch_policy.py:413]: done. I found that it should actually be torch.cat.

File: rllib/algorithms/crr/torch/crr_torch_policy.py (hunk 3 above)
Comment (gjoliver) [reply rllib/algorithms/crr/torch/crr_torch_policy.py:417]: done

File: rllib/algorithms/crr/torch/crr_torch_policy.py (hunk 2 above)
Comment (gjoliver) [reply rllib/algorithms/crr/torch/crr_torch_policy.py:410]: done. let me know if you think the names are still not good.

**Review by kouroshHakha — APPROVED** [review]

**Comment by gjoliver** [comment]
merging, R2D2 failure is not related.
